src/serialized-analysis.ts

```diff
@@ -0,0 +1,425 @@
+   */
+  demos: string[];
+
+  /**
+   * The tagname that the element extends, if any. The value of the `extends`
+   * option that's passed into `customElements.define`.
+   *
+   * e.g. `input`, `paper-button`, `my-super-element`
+   */
+  extends?: string;
+
+  /**
+   * The class name for this element.
+   *
+   * e.g. `MyElement`, `Polymer.PaperInput`
+   */
+  classname: string;
+
+  /**
+   * The class that this element extends.
+   *
+   * This is non-optional, as every custom element must have HTMLElement in
+   * its prototype change.
+   *
+   * e.g. `HTMLElement`, `HTMLInputElement`, `MyNamespace.MyBaseElement`
+   */
+  superclass: string;
+
+  /** The attributes that this element is known to understand. */
+  attributes?: Attribute[];
+
+  /** The properties that this element has. */
+  properties?: Property[];
+
+  /** The events that this element fires. */
+  events?: Event[];
+
+  /** The shadow dom content slots that this element accepts. */
+  'slots': Slot[];
+
+  /** Information useful for styling the element and its children. */
+  styling: {
+
+    /** CSS Classes that produce built-in custom styling for the element. */
+    classes: {
+      /** The name of the class. e.g. `bright`, `cascade`, `ominous_pulsing` */
+      name: string;
+      /** A markdown description of the class and what effects it has. */
+      description: string;
+    }[];
+
+    /** CSS Variables that the element understands. */
+    cssVariables: {
+
+      /** The name of the variable. e.g. `--header-color`, `--my-element-size`*/
+      name: string;
+
+      /** The type of the variable. Advisory. e.g. `color`, `size` */
+      type?: string;
+
+      /** A markdown description of the variable. */
+      description?: string;
+
+      /**
+       * A markdown description of how the element will fallback if the variable
+       * isn't defined.
+       */
+      fallbackBehavior?: string;
+    }[];
+
+    /** If true, the element must be given an explicit size by its context. */
+    needsExplicitSize?: boolean;
+
+    // Would be nice to document the default styling a bit here, whether it's
+    // display: block or inline or whatever.
+  };
+
+  /**
+   * An extension point for framework-specific metadata, as well as any
+   * metadata not yet standardized here such as what polyfills are needed,
+   * behaviors and mixins used, the framework that the element was written in,
+   * tags/categories, links to specs that the element implements, etc.
+   */
+  metadata: Object;
+}
+
+export interface Attribute {
+  /** The name of the attribute. e.g. `value`, `icon`, `should-collapse`. */
+  name: string;
+
+  /** A markdown description for the attribute. */
+  description?: string;
+
+  /**
+   * The type that the attribute will be serialized/deserialized as.
+   *
+   * e.g. `string`, `number`, `boolean`, `RegExp`, `Array`, `Object`.
+   */
+  type?: string;
+
+  /** The default value of the attribute, if any. */
+  defaultValue?: string;
```

Comment:
We need some way to indicate if an attribute is associated with a property... TBD?